geoips/models/v2alpha1/bases.py

```diff
@@ -0,0 +1,403 @@
+    are allowed, and the object data can be modified after initialization.
+    """
+
+    model_config = ConfigDict(extra="allow", populate_by_name=True, frozen=False)
+
+
+def python_identifier(val: str) -> str:
+    """Validate if a string is a valid Python identifier.
+
+    Validate if a string is a valid Python identifier and not a reserved Python keyword.
+    `See <https://docs.python.org/3/reference/lexical_analysis.html#identifiers>`_ for
+    more information on Python identifiers and reserved keywords.
+
+    Validation is performed by calling `str.isidentifier` and `keyword.iskeyword`.
+
+    Parameters
+    ----------
+    val : str
+        The input string to validate.
+
+    Returns
+    -------
+    str
+        The input string if it is a valid Python identifier.
+
+    Raises
+    ------
+    ValueError
+        If the input string is invalid as a Python identifier or a reserved keyword.
+    """
+    error_messages = []
+    if not val.isidentifier():
+        error_messages.append(f"'{val}' is not a valid Python identifier.")
+    if keyword.iskeyword(val):
+        error_messages.append(f"'{val}' is a reserved Python keyword.")
+
+    if error_messages:
+        error_message = " ".join(error_messages) + " Please update it."
+        LOG.interactive(error_message, exc_info=True)
+        raise ValueError(error_message)
+
+    return val
+
+
+# Create the PythonIdentifier type
+PythonIdentifier = Annotated[str, AfterValidator(python_identifier)]
+
+
+def get_interfaces(namespace) -> set[str]:
+    """Return a set of distinct interfaces.
+
+    This function returns all available plugin interfaces. The results are cached for
+    runtime memory optimization.
+
+    Returns
+    -------
+    set of str
+        set of interfaces
+    """
+    if namespace == "geoips.plugin_packages":
+        return {
+            available_interfaces
+            for ifs in interfaces.list_available_interfaces().values()
+            for available_interfaces in ifs
+        }
+    else:
+        mod = get_interface_module(namespace)
+        return set(mod.__all__)
+
+
+class PluginModelMetadata(ModelMetaclass):
+    """API version and namespace metadata for the corresponding plugin model.
+
+    This is used to derive 'apiVersion' and 'namespace' for any given PluginModel.
+    PluginModel can be instantiated directly or a child class of PluginModel can be
+    instantiated and the functionality will for the same.
+
+    Initially attempted to use __init_subclass__ in the PluginModel class itself, but
+    that only supported child classes of PluginModel (i.e. WorkflowPluginModel, ...),
+    but not instantiation of PluginModel itself.
+
+    NOTE: Need to inherit from ModelMetaclass, otherwise we'll wind up with this error:
+
+    E TypeError: metaclass conflict: the metaclass of a derived class must be a
+    (non-strict) subclass of the metaclasses of all its bases
+    """
+
+    def __new__(mcs, name, bases, namespace, **kwargs):
```

Comment:
Not something we necessarily need to fix, but just flagging that this code has inconsistent use of type hints on some functions but not others.